src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -212,19 +212,31 @@ public function denormalize(mixed $data, string $type, string $format = null, ar
             throw new NotNormalizableValueException(sprintf('Could not denormalize object of type "%s", no supporting normalizer found.', $type));
         }
 
+        $notNormalizableExceptions = [];
         if (isset($context[DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS])) {
+            if ($context[DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS]) {
+                $context['not_normalizable_value_exceptions'] = [];
+                $notNormalizableExceptions = &$context['not_normalizable_value_exceptions'];
+            }
             unset($context[DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS]);
-            $context['not_normalizable_value_exceptions'] = [];
-            $errors = &$context['not_normalizable_value_exceptions'];
-            $denormalized = $normalizer->denormalize($data, $type, $format, $context);
-            if ($errors) {
-                throw new PartialDenormalizationException($denormalized, $errors);
+        }
+
+        $extraAttributesExceptions = [];
+        if (isset($context[DenormalizerInterface::COLLECT_EXTRA_ATTRIBUTES_ERRORS])) {
+            if ($context[DenormalizerInterface::COLLECT_EXTRA_ATTRIBUTES_ERRORS]) {
+                $context['extra_attributes_exceptions'] = [];
+                $extraAttributesExceptions = &$context['extra_attributes_exceptions'];
             }
+            unset($context[DenormalizerInterface::COLLECT_EXTRA_ATTRIBUTES_ERRORS]);
+        }
+
+        $denormalized = $normalizer->denormalize($data, $type, $format, $context);
 
-            return $denormalized;
+        if (\count($notNormalizableExceptions) > 0 || \count($extraAttributesExceptions) > 0) {
+            throw new PartialDenormalizationException($denormalized, $notNormalizableExceptions ?? [], $extraAttributesExceptions ?? []);
```

Comment:
```suggestion
            throw new PartialDenormalizationException($denormalized, $notNormalizableExceptions, $extraAttributesExceptions);
```
As these variables will always be set